pennylane/qnode.py

```diff
@@ -670,15 +670,21 @@ def __call__(self, *args, **kwargs):
             # construct the tape
             self.construct(args, kwargs)
 
-        # If the observable contains a Hamiltonian and the device does not
-        # support Hamiltonians, or if the simulation uses finite shots,
-        # split tape into multiple tapes of diagonalizable known observables.
-        # In future, this logic should be moved to the device
-        # to allow for more efficient batch execution.
+        # Under certain conditions, split tape into multiple tapes and recombine them.
+        # Else just execute the tape, and let the device take care of things.
+        # TODO (future squad): This logic should be moved away from the qnode, preferably to the device.
+        hamiltonian_in_obs = "Hamiltonian" in [obs.name for obs in self.qtape.observables]
+        # if the device does not support Hamiltonians, we split them

```

Comment:
We can already do this 🙂 If using `qml.beta.qnode`, it simply devolves the expansion to the device, here:

https://github.com/PennyLaneAI/pennylane/blob/c9e044cd602bb80379d08686f28014dd8ecf7674/pennylane/_device.py#L641

Reply:
I know :) 

But I fear we need the device refactor, because that logic should be default for all devices...